### PR TITLE
Add functionality to load code from a query param

### PR DESCRIPTION
**Pull request by ZeWaka**
also load the first downloaded version by default

### CODE REVIEW

File: src/app/components/editor/editor.component.ts [hunk 1]
```diff
@@ -82,6 +83,23 @@ export class EditorComponent {
         flexGrow: 1,
       },
     });
+
+    // Check for URL parameter and decode if present
+    this.route.queryParams.subscribe(params => {
+      const encodedContent = params['code'];
+      if (encodedContent) {
+        const decodedContent = atob(encodedContent);
```
Comment (alexkar598) [inline src/app/components/editor/editor.component.ts:91]: what happens if this throws can you test if it still responds to changes or does it unsubscribe

File: src/vm/byond.service.ts [hunk 2]
```diff
@@ -131,6 +131,10 @@ export class ByondService {
     void this.lock.run(async () => {
       for await (const version of (await this.getByondFolder()).keys()) {
         this._versions.set(version, VersionStatus.Fetched);
+        // If this is the first version in the list, load it
+        if (this._activeVersion == null) {
+          await this.load(version, true);
```
Comment (alexkar598) [inline src/vm/byond.service.ts:136]: I'm not sure how meaningful the status of first installed version is. In fact, it's probably the oldest version and the one the user is the least likely to want

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: src/vm/byond.service.ts (hunk 2 above)
Comment (ZeWaka) [reply src/vm/byond.service.ts:136]: i just want /some/ version to be actively loaded
so you don't have to click 'set active' every single time

the user can manage their own versions

File: src/app/components/editor/editor.component.ts (hunk 1 above)
Comment (ZeWaka) [reply src/app/components/editor/editor.component.ts:91]: image

**Pull request merged**